ena_portal_api/ena_handler.py

```diff
@@ -55,26 +66,30 @@ def post_request(self, data):
             response = requests.post(self.url, data=data, **get_default_connection_headers())
         return response
 
-    def get_study(self, study_sec_acc):
+    # Supports ENA primary and secondary study accessions
+    def get_study(self, study_acc):
         data = get_default_params()
         data['result'] = 'study'
         data['fields'] = 'study_accession,secondary_study_accession,study_description,study_name,study_title,' \
                          'center_name,broker_name,last_updated,first_public'
-        data['query'] = 'secondary_study_accession=\"{}\"'.format(study_sec_acc)
+        if study_acc[0:2] in ('ERP', 'SRP', 'DRP'):
+            data['query'] = 'study_accession=\"{}\"'.format(study_acc)
+        else:
+            data['query'] = 'secondary_study_accession=\"{}\"'.format(study_acc)
         response = self.post_request(data)
         if str(response.status_code)[0] != '2':
-            raise ValueError('Could not retrieve runs for study %s.', study_sec_acc)
+            raise ValueError('Could not retrieve runs for study %s.', study_acc)
         try:
             study = json.loads(response.text)[0]
         except IndexError:
-            raise IndexError('Could not find study {} in ENA.'.format(study_sec_acc))
+            raise IndexError('Could not find study {} in ENA.'.format(study_acc))
         return study
 
     def get_runs(self, run_accession):
         data = get_default_params()
         data['result'] = 'read_run'
         data['fields'] = 'secondary_study_accession,run_accession,library_source,library_strategy,' \
-                         'library_layout,fastq_ftp,base_count,read_count,instrument_platform,instrument_model',
+                         'library_layout,fastq_ftp,base_count,read_count,instrument_platform,instrument_model,secondary_sample_accession',
```

Comment:
This might be useful to run an md5 checksum after the download finished.
```suggestion
                         'library_layout,fastq_md5,fastq_ftp,base_count,read_count,instrument_platform,instrument_model,secondary_sample_accession',
```